src/patch_parser.py

```diff
@@ -414,105 +422,97 @@ def _does_match_end_of_super_chunk(lines: InputSource) -> bool:
 
 def _parse_patch_file_unchanged_chunk(
         lines: InputSource,
-        gerrit_orig_line: int,
-        gerrit_new_line: int) -> Tuple[int, int, PatchFileChunkLineMap]:
+        parser_state: HunkParserState) -> PatchFileChunkLineMap:
     in_start = lines.line_number()
     while (not _does_match_end_of_super_chunk(lines)) and (
             (not lines[0]) or (lines[0] and lines[0][0] != '+' and lines[0][0] != '-')):
         logging.info('dropping line: %s', lines[0])
         lines.consume()
-        gerrit_orig_line += 1
-        gerrit_new_line += 1
-    return (gerrit_orig_line,
-            gerrit_new_line,
-            PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
-                                  side='',
-                                  offset=(gerrit_new_line - lines.line_number())))
+        parser_state.gerrit_orig_line += 1
+        parser_state.gerrit_new_line += 1
+        # logging.debug('Unchanged start: ' + str(in_start))
+        # logging.debug('Unchanged line: ' + lines[0])
+        # logging.debug('Unchanged lines - 1: ' + str(lines.line_number() - 1))
+    offset = parser_state.gerrit_new_line - parser_state.deleted_lines - lines.line_number()
+    return PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
+                                 side='', offset=offset)
 
 
 def _parse_patch_file_added_chunk(
         lines: InputSource,
-        gerrit_orig_line: int,
-        gerrit_new_line: int) -> Tuple[int, int, PatchFileChunkLineMap]:
+        parser_state: HunkParserState) -> PatchFileChunkLineMap:
     in_start = lines.line_number()
+    previous_number_of_removed_lines = parser_state.deleted_lines
```

Comment:
this var is currently unused, did you want to use it to compute offset?
i.e.
offset=(parser_state.gerrit_new_line - lines.line_number() - previous_deleted_lines
?

or that can be left for another commit and we can drop this var for now.